File: 프로그래머스/2/12911. 다음 큰 숫자/다음 큰 숫자.py
def solution(n):
    oneCount = len([i for i in list(format(n, 'b')) if i == '1']);
    
    result = n + 1;
    while True:
        temp = len([i for i in list(format(result, 'b')) if i == '1']);
        if oneCount == temp: return result
        else: result += 1

File: 프로그래머스/2/12911. 다음 큰 숫자/다음 큰 숫자.js
function solution(n) {
    // const oneCount = n.toString(2).split('').filter(el => el === '1').length;
    const oneCount = n.toString(2).split('').reduce((acc, cur) => cur === '1' ? acc + 1 : acc, 0);
    let result = n + 1;
    
    while(result > n){
        let temp = result.toString(2).split('').filter(el => el === '1').length;
        if(oneCount === temp) return result
        else result++
    }
}